hclsyntax/expression_test.go

```diff
@@ -342,6 +342,56 @@ upper(
 			cty.DynamicVal,
 			0,
 		},
+		{
+			`foo::upper("foo")`,
+			&hcl.EvalContext{
+				Functions: map[string]function.Function{
+					"foo::upper": stdlib.UpperFunc,
+				},
+			},
+			cty.StringVal("FOO"),
+			0,
+		},
+		{
+			`foo :: upper("foo")`, // spaces are non-idomatic, but valid
```

Comment:
Is there a strong reason to allow spaces like this? Given that the way namespaces are exposed - as part of the opaque `Name` of `FunctionCallExpr` - this could make it more difficult to interpret it - downstream will likely have to do some whitespace trimming before doing anything with the name.

Reply:
@dbanck just helpfully pointed out that the name which becomes part of the expression type is already space-trimmed - which I guess makes it slightly better, but I would still ask why is it necessary to allow the spaces?

The language as a whole already allows a number of non-idiomatic choices - which seems reasonable if they weren't originally assumed to be non-idiomatic. It does feel odd to intentionally introduce new feature and intentionally allow non-idiomatic code though.

Reply:
Understood, thanks for the context.